extras/wait-for-bricks.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+#Copyright (c) 2018 Red Hat, Inc. <http://www.redhat.com>
+#This file is part of GlusterFS.
+#
+#This file is licensed to you under your choice of the GNU Lesser
+#General Public License, version 3 or any later version (LGPLv3 or
+#later), or the GNU General Public License, version 2 (GPLv2), in all
+#cases as published by the Free Software Foundation.
+
+
+function volinfo_field()
+{
+    local vol=$1;
+    local field=$2;
+
+    gluster volume info $vol --xml| grep -oPm1 "(?<=<$field>)[^<]+"
+}
+
+function is_enabled()
+{
+        case $1 in
+        0|off|no|false|disable)
+                echo "N"
+                ;;
+        1|on|yes|true|enable)
+                echo "Y"
+                ;;
+        *)
+                echo "N/A"
+                ;;
+        esac
+}
+
+function volinfo_option()
+{
+    local vol=$1;
+    local key=$2;
+
+    gluster volume get $vol $key| awk '{print $NF}' | tail -1
+}
+
+function expected_online_brick_count()
+{
+        if [ "0" == $(volinfo_field "all" "count") ];
+        then
+                echo 0
+                return 0
+        fi
+        gluster volume list | while read volname;
+        do
+                if [ "Started" == $(volinfo_field $volname 'statusStr') ];
+                then
+                        if [ "Y" == $(is_enabled $(volinfo_option $volname features.shard)) ];
+                        then
+                                brick_count=$(volinfo_field $volname 'brickCount')
+                                number_of_bricks=$((number_of_bricks + brick_count))
+                        fi
+                fi
+
+                if [ -z $number_of_bricks ];
+                then
+                        echo 0
+                else
+                        echo $number_of_bricks
+                fi
+        done | tail -1
+}
+
+function check_brick_status() {
+        local volname=$1
+        local daemon=$2
+
+        cmd="gluster --xml volume status $volname"
+        if [[ -z $daemon ]]
```

Comment:
If we want to use [[    ]]  style lets keep it consistent through out the file. But from my past experience we better use [[   ]] instead of [  ]